### PR TITLE
Add Fedora 22 Ganeti packer definitions

### PR DESCRIPTION
**Pull request by borcean**
Adding support for building Fedora 22 images for Ganeti with Packer.

### CODE REVIEW

**Comment by borcean** [comment]
@ramereth Can you take a look at this when you have a chance? 😊

File: packer/scripts/fedora/ganeti.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash -eux
+
+if [ -e /usr/bin/dnf ] ; then
```
Comment (jordane) [inline packer/scripts/fedora/ganeti.sh]: `-x` is probably better than `-e` here.

Comment (borcean) [inline packer/scripts/fedora/ganeti.sh]: @jordane Is there a reason to check execution over existence? All the others are configured to just check existence.

File: packer/fedora-22-x86_64-ganeti.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      "type": "qemu",
+      "vm_name": "packer-fedora-22-x86_64"
+    }
+  ],
+  "provisioners": [
+    {
+      "environment_vars": [
+        "CHEF_VERSION={{user `chef_version`}}"
+      ],
+      "execute_command": "echo 'vagrant' | {{.Vars}} sudo -E -S bash '{{.Path}}'",
```
Comment (ramereth) [inline packer/fedora-22-x86_64-ganeti.json]: This should be 'osuadmin', not 'vagrant'

File: packer/scripts/fedora/ganeti.sh (hunk 1 above)
Comment (jordane) [inline packer/scripts/fedora/ganeti.sh]: @borcean not a terribly good one, but we do intend to execute the file if it exists.

File: packer/scripts/fedora/ganeti.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+if [ -x /usr/bin/dnf ] ; then
+  dnf -y install cloud-init cloud-utils yum
+else
+  yum -y install cloud-init cloud-utils dracut-modules-growroot
+fi
+dracut -f
+
+if [ -e /boot/grub/grub.conf ] ; then
+  sed -i -e 's/rhgb.*/console=ttyS0,115200n8 console=tty0 quiet/' /boot/grub/grub.conf
```
Comment (ramereth) [inline packer/scripts/fedora/ganeti.sh]: We don't need to redirect the console to serial on Ganeti.

File: packer/scripts/fedora/ganeti.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  sed -i -e 's/^timeout=.*/timeout=0/' /boot/grub/grub.conf
+  cd /boot
+  ln -s boot .
+elif [ -e /etc/default/grub ] ; then
+  # output bootup to serial
+  sed -i -e \
+    's/GRUB_CMDLINE_LINUX=\"\(.*\)/GRUB_CMDLINE_LINUX=\"console=ttyS0,115200n8 console=tty0 \1/g' \
+    /etc/default/grub
```
Comment (ramereth) [inline packer/scripts/fedora/ganeti.sh]: We don't need to redirect the console to serial on Ganeti.

File: packer/http/fedora-22/ks-ganeti.cfg [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+nfs-utils
+rsync
+tar
+vim
+wget
+%end
+
+%post
+# update root certs
+wget -O/etc/pki/tls/certs/ca-bundle.crt http://curl.haxx.se/ca/cacert.pem
```
Comment (jordane) [inline packer/http/fedora-22/ks-ganeti.cfg]: Is there a reason to not grab this straight from mozilla?

Also, the certs in f22 should be quite up to date.

Comment (ramereth) [inline packer/http/fedora-22/ks-ganeti.cfg]: This was an artifact from pulling how the upstream bento does it.

Comment (borcean) [inline packer/http/fedora-22/ks-ganeti.cfg]: @ramereth Is this something that should be changed?

Comment (ramereth) [inline packer/http/fedora-22/ks-ganeti.cfg]: It's debatable. I've been trying to keep this aligned with upstream bento within reason. What's the downside of keeping this?

Comment (jordane) [inline packer/http/fedora-22/ks-ganeti.cfg]: The downside is that we're now trusting `haxx.se` with our root ssl certs...

Comment (ramereth) [inline packer/http/fedora-22/ks-ganeti.cfg]: Let's fix this here then to pull from mozilla, but make an issue/PR to change it globally on all our openstack/ganeti builds.

Comment (jordane) [inline packer/http/fedora-22/ks-ganeti.cfg]: @borcean please pull this from https://hg.mozilla.org/releases/mozilla-release/raw-file/default/security/nss/lib/ckfw/builtins/certdata.txt and make sure to use https.

Comment (borcean) [inline packer/http/fedora-22/ks-ganeti.cfg]: @ramereth I don't have the permission to open an issue in the bento repo. This one has been updated, and I will get a PR for removing it globally tomorrow.

Comment (jordane) [inline packer/http/fedora-22/ks-ganeti.cfg]: I added issues to the repo, so issues can now be opened.

**Comment by ramereth** [comment]
Looks good! Great work :smile: :+1:

**Pull request merged**